### PR TITLE
Refactored versioning directives to take in a default value.

### PR DESCRIPTION
**Pull request by kshakir**
If one does not want to pass a default, use the new directives with "Optional" in their names.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/vault/common/directives/VersioningDirectives.scala [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
-import spray.routing.{PathMatcher, Directive}
+import shapeless.{::, HList, HNil}
 import spray.routing.directives.BasicDirectives._
 import spray.routing.directives.PathDirectives._
+import spray.routing.{Directive, PathMatcher}
 
 object VersioningDirectives {
-  val optionalVersion: Directive[Option[Int] :: HNil] =
-    pathPrefix("v" ~ IntNumber).map(Option(_)) | provide(None)
+  val optionalVersion: Directive[Option[Int] :: HNil] = pathPrefix("v" ~ IntNumber).map(Option(_)) | provide(None)
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/vault/common/directives/VersioningDirectives.scala:63]: Please document and provide usage examples for each of these.

Are any intended for internal use only?  Mark as private/protected as appropriate.

Comment (kshakir) [inline src/main/scala/org/broadinstitute/dsde/vault/common/directives/VersioningDirectives.scala:63]: For examples, is it ok if I point users to the tests? Or do you prefer them to be inline?

At minimum, will add brief scaladocs to each.

Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/vault/common/directives/VersioningDirectives.scala:63]: I would like a brief discussion of `path` vs. `pathPrefix` and `version` vs. `optionalVersion`, as well as why some of these have overloads for `PathMatcher` but others do not.

**Comment by kshakir** [comment]
Basic docs added. In any sections needing clarity / correction, please feel free to edit or let me know.

**Comment by jmthibault79** [comment]
Excellent docs!  Thanks.

A tiny change, then squash and merge.

**Pull request merged**